bindings/java/src/main/com/apple/foundationdb/FDB.java

```diff
@@ -167,23 +167,36 @@ public static FDB instance() throws FDBException {
 	 *  object.<br><br>
 	 *
 	 *  Warning: When using the multi-version client API, setting an API version that
-	 *   is not supported by a particular client library will prevent that client from 
+	 *   is not supported by a particular client library will prevent that client from
 	 *   being used to connect to the cluster. In particular, you should not advance
-	 *   the API version of your application after upgrading your client until the 
+	 *   the API version of your application after upgrading your client until the
 	 *   cluster has also been upgraded.
 	 *
+	 *  Note: automatically stops network thread upon runtime shutdown, if this
+	 *   behavior is undesirable use {@link #selectAPIVersion(int, boolean)}
+	 *
 	 * @param version the API version required
 	 *
 	 * @return the FoundationDB API object
+	 * @see #selectAPIVersion(int, boolean)
 	 */
 	public static FDB selectAPIVersion(final int version) throws FDBException {
 		return selectAPIVersion(version, true);
 	}
 
 	/**
-	   This function is called from C++ if the VM is controlled directly from FDB
+	 * Same as {@link #selectAPIVersion(int)} but lets use choose whether you want to close
+	 * FDB connection automatically on runtime shutdown or you want to do it manually.
+	 *
+	 * Dev Note: This function is called from C++ for tests, method signature must be consistent
+	 * with JNI call.
+	 *
+	 * @param version the API version required
+	 * @param stopOnShutdown whether you need to stop network thread upon Java runtime shutdown (via
+	 *  Runtime.getRuntime().addShutdownHook)
+	 * @see #selectAPIVersion(int)
 	 */
-	private static synchronized FDB selectAPIVersion(final int version, boolean controlRuntime) throws FDBException {
+	public static synchronized FDB selectAPIVersion(final int version, boolean stopOnShutdown) throws FDBException {
```

Comment:
I see this function existed previously as a private function, but I think the effect here is orthogonal to what `selectAPIVersion` is intended to do. If we are going to make the ability to control the shutdown hook public, I think it would be cleaner to have a function to explicitly disable it (e.g. `FDB.disableShutdownHook`, or even something more generic such that you allow custom shutdown hooks) and then move the hook creation to `startNetwork` where we can check if it's enabled. So you would do something like this:

```
FDB fdb = FDB.selectAPIVersion(700);
fdb.disableShutdownHook();
Database db = fdb.open();
```

Reply:
Thank you for your attention. Totally agree that `selectAPIVersion` does not look the right place for this, but I was trying to make least invasive change possible. I will change the API in a way you suggest.

P.S.: I'm not sure there is need for custom shutdown hook because all it does at shutdown just calling FDB::stopNetwork which is public and so can be easily called manually.

Reply:
> P.S.: I'm not sure there is need for custom shutdown hook because all it does at shutdown just calling FDB::stopNetwork which is public and so can be easily called manually.

Yeah, good point. You could just incorporate `stopNetwork` into your own shutdown hook in whatever way is appropriate, so there's nothing fancy needed from our API for this.

Reply:
@ajbeamon : added disableShutdownHook and synchronized with JavaWorkload.cpp code. Unfortunately, I did not found any java workload tests inside fdb/tests.